src/trunk.c

```diff
@@ -6205,13 +6206,11 @@ trunk_lookup(trunk_handle *spl, char *key, writable_buffer *data)
       platform_status rc;
       rc = trunk_memtable_lookup(spl, mt_gen, key, data);
       platform_assert_status_ok(rc);
-      if (!writable_buffer_is_null(data)) {
-         slice message = writable_buffer_to_slice(data);
-         type          = data_message_class(data_cfg, message);
-         if (type != MESSAGE_TYPE_UPDATE) {
-            found_in_memtable = TRUE;
-            goto found_final_answer_early;
-         }
+      if (!merge_accumulator_is_null(data)
```

Comment:
Maybe a wrapper function that conveys the higher-level meaning of this? Something like `trunk_lookup_found(result)`? (including the renaming I suggest above)

Reply:
Then this whole statement reads as
```
if (trunk_lookup_found(result) && message_is_final(result))
```
or something similar.